packages/ui/src/pages/boardgame/generateSchemaValidation.ts

```diff
@@ -0,0 +1,54 @@
+import { Property } from '@/pages/boardgame/components/property-form'
+import { isNil } from '@/utils'
+
+export const generateSchemaFromProperties = (properties: {
+  [k: string]: Property | null
+}) => {
+  const formattedProperties = Object.values(properties).reduce(
+    (acc, property) => {
+      if (isNil(property)) {
+        return acc
+      }
+      return {
+        ...acc,
+        [property.name]: { type: property.type.toLowerCase() },
+      }
+    },
+    {
+      player: {
+        type: 'string',
+        pattern:
+          '^[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}$',
+      },
+    },
+  )
+  const propertyArray = Object.keys(formattedProperties)
+    .map(name => `"${name}"`)
+    .toString()
+
+  return `
```

Comment:
Could this be a stringified object instead of a string

Reply:
that is actually better, and easier to test :D!